src/components/common/Layout/components/Account/components/AccountModal/components/SocialAccount/FarcasterProfile.tsx

```diff
@@ -0,0 +1,318 @@
+    })
+    onSuccess?.()
+  }
+
+  const submitSignedKeyRequest = (
+    signedPayload: SignedValidation
+  ): Promise<{ url: string; deadline: number; deadlineRelative: number }> =>
+    fetcher(`/v2/users/${userId}/farcaster-signed-keys`, signedPayload)
+
+  const onRegenerate = () => {
+    signedKeyRequest.onSubmit()
+  }
+
+  const { seconds, start, stop } = useCountdownSeconds(undefined, () => {
+    captureEvent("[farcaster] deadline hit")
+    onRegenerate()
+  })
+  const shouldEnableRegenerateButton = seconds < ENABLE_REGENERATE_BUTTON_AT_SEC
+
+  const signedKeyRequest = useSubmitWithSign(submitSignedKeyRequest, {
+    onSuccess: ({ deadlineRelative }) => {
+      const deadline = Date.now() + deadlineRelative
+      start(Math.floor(deadlineRelative / 1000) - POLL_EARLY_END_SEC) // 2 minuter earlier just for good measure
+
+      if (pollInterval) {
+        clearInterval(pollInterval)
+      }
+
+      setTimeout(() => {
+        const interval = setInterval(() => {
+          if (Date.now() > deadline) {
+            clearInterval(interval)
+          }
+
+          fetcherWithSign([
+            `/v2/users/${userId}/farcaster-profiles`,
+            { method: "GET" },
+          ])
+            .then((profiles) => {
+              if (profiles?.length > 0) {
+                mutate(
+                  (prev) => (prev ? { ...prev, farcasterProfiles: profiles } : prev),
+                  {
+                    revalidate: false,
+                  }
+                ).then(() => {
+                  onApprove()
+                  clearInterval(interval)
+                })
+              }
+            })
+            .catch(() => {})
+        }, APPROVAL_POLL_INTERVAL_MS)
+
+        setPollInterval(interval)
+      }, APPROVAL_POLL_INITIAL_DELAY_MS)
+
+      onOpen()
+    },
+  })
+
+  const handleOnClose = () => {
+    captureEvent("[farcaster] connect modal closed")
+    stop()
+    signedKeyRequest.reset()
+    if (pollInterval) {
+      clearInterval(pollInterval)
+    }
+    onClose()
+  }
+
+  const qrSize = useBreakpointValue({ base: 300, md: 400 })
+
+  return (
+    <>
+      <Button
+        onClick={() => {
+          captureEvent("[farcaster] connect button clicked")
+          signedKeyRequest.onSubmit()
+        }}
+        colorScheme={rewards.FARCASTER.colorScheme}
+        variant={"solid"}
+        size="sm"
+        isDisabled={farcasterProfiles?.length > 0}
+        isLoading={signedKeyRequest.isLoading}
+        {...props}
+      >
+        Connect
+      </Button>
+      <Modal isOpen={isOpen} onClose={handleOnClose} size="lg">
+        <ModalOverlay />
+        <ModalContent>
+          <ModalCloseButton />
+          <ModalHeader pb={0} pr={16} display={"flex"} gap={1} alignItems={"center"}>
+            {isMobile ? (
+              "Connect Farcaster"
+            ) : (
+              <>
+                <Icon boxSize={6} as={DeviceMobileCamera} /> Scan With your phone
```

Comment:
We might need to wrap the text in a span, not sure if translator plugins can handle the title this way

Reply:
Thanks for pointing that out! Wrapped in a Text in https://github.com/guildxyz/guild.xyz/pull/1301/commits/0f766935d25a9b2969fea9d53b82d569a256b691